widget/embedded/src/components/SwapDetails/SwapDetails.Modal.tsx

```diff
@@ -0,0 +1,137 @@
+            ? i18n.t('Changing Network')
+            : PendingSwapNetworkStatus.WaitingForConnectingWallet
+            ? i18n.t('Connected Wallet')
+            : i18n.t('Network Changed')
+        }
+        description={message}
+      />
+      {currentStepWallet?.walletType && (
+        <div
+          style={{
+            display: 'flex',
+            justifyContent: 'center',
+          }}>
+          <Wallet
+            title={getWalletInfo(currentStepWallet.walletType).name}
+            image={getWalletInfo(currentStepWallet.walletType).img}
+            type={currentStepWallet?.walletType}
+            state={
+              !walletState(currentStepWallet.walletType).connected
+                ? WalletState.DISCONNECTED
+                : WalletState.CONNECTED
+            }
+            onClick={async () => connect(currentStepWallet.walletType)}
+          />
+        </div>
+      )}
+    </>
+  );
+};
+
+const DeleteCancelContent = ({
+  deleteAction,
+  onClose,
+  isDelete,
+  onCancel,
+}: DeleteCancelContentProps) => {
+  return (
+    <>
+      <Divider size={20} />
+      <MessageBox
+        type="error"
+        title={isDelete ? i18n.t('Delete Transaction') : 'Transaction Cancel'}
+        description={
+          isDelete
+            ? i18n.t('Are you sure you want to delete this swap?')
+            : i18n.t('Are you sure you want to Cancel this transaction?')
+        }
+      />
+      <Divider size={32} />
+      <Button
+        variant="contained"
+        type="primary"
+        size="large"
+        onClick={isDelete ? deleteAction : onCancel}>
+        <Typography variant="title" size="medium" color="neutral100">
+          {isDelete ? i18n.t('Yes, Delete') : i18n.t('Yes, Cancel transaction')}
+        </Typography>
+      </Button>
+      <Divider size={12} />
+      <Button variant="outlined" type="primary" size="large" onClick={onClose}>
+        <Typography variant="title" size="medium" color="primary">
+          {isDelete ? i18n.t('No, Cancel') : i18n.t('No, Continue')}
+        </Typography>
+      </Button>
+      <Divider size={20} />
+      <BottomLogo />
+    </>
+  );
+};
+
+export function ModalSwapDetails(props: ModalPropTypes) {
+  const { state, onClose, currentStepWallet, message, ...restProps } = props;
+
+  return (
+    <Modal
+      open={!!state}
+      onClose={onClose}
+      container={document.getElementById('swap-box') || document.body}>
+      {[
+        PendingSwapNetworkStatus.WaitingForNetworkChange,
+        PendingSwapNetworkStatus.WaitingForConnectingWallet,
+        PendingSwapNetworkStatus.WaitingForConnectingWallet,
+      ].includes(state as PendingSwapNetworkStatus) && (
+        <WalletStateContent
+          state={state}
+          currentStepWallet={currentStepWallet}
+          message={message}
+        />
+      )}
+      {state === 'delete' && (
+        <DeleteCancelContent isDelete={true} onClose={onClose} {...restProps} />
```

Comment:
Is there any reason that you have merged these two contents together? We could have `DeleteContent` and `CancelContent` components to avoid unnecessary conditions.

Reply:
fixed.

Reply:
fixed.